source/defclass-star.lisp

```diff
@@ -253,29 +265,40 @@ Use the slot name directly:
         (*symbols-to-export* nil)
         (*export-class-name-p* export-class-name)
         (*export-accessor-names-p* export-accessor-names)
-        (*export-slot-names-p* export-slot-names))
+        (*export-slot-names-p* export-slot-names)
+        (*export-predicate-name-p* export-predicate-name)
+        (*predicate-name-transformer* predicate-name-transformer))
     (multiple-value-bind (binding-names binding-values clean-options)
         (extract-options-into-bindings options)
       (progv binding-names (mapcar #'eval binding-values)
         (let ((result (funcall expansion-builder
                                (mapcar 'process-slot-definition slots)
                                clean-options)))
-          (if (or *symbols-to-export*
-                  *export-class-name-p*)
-              `(progn
-                 ,result
-                 (eval-when (:compile-toplevel :load-toplevel :execute)
-                   ;; Don't try to export symbols that don't belong to *package*.
-                   ;; This can happen when inheriting from a class and
-                   ;; overriding some slot.
-                   (export '(,@(remove-if (lambda (sym)
-                                            (not (eq (symbol-package sym) *package*)))
-                                (append (when *export-class-name-p*
-                                          (list name))
-                                 *symbols-to-export*)))
-                           ,(package-name *package*)))
-                 (find-class ',name nil))
-              result))))))
+          `(progn
+             ,@(when *predicate-name-transformer*
+                 (let ((pred-name (funcall *predicate-name-transformer* name)))
+                   `((defun ,pred-name (object)
+                       (subtypep (type-of object) ',name))
+                     ,@(when *export-predicate-name-p*
+                         `((eval-when (:compile-toplevel :load-toplevel :execute)
+                             (export ',pred-name
+                                     ,(package-name *package*))))))))
+             ,(if (or *symbols-to-export*
+                      *export-class-name-p*)
+                  `(progn
+                     ,result
+                     (eval-when (:compile-toplevel :load-toplevel :execute)
+                       ;; Don't try to export symbols that don't belong to *package*.
+                       ;; This can happen when inheriting from a class and
+                       ;; overriding some slot.
+                       (export '(,@(remove-if (lambda (sym)
```

Comment:
as you can see, this random exporting from `*package*` requires consideration. maybe even the entire package would benefit from a review. (i won't do that any time soon)

Reply:
Note that this portion was fixed with 39d458f1b1bc830d1f5e18a6a35bf0e96a2cfd61, it's not port of this pull request.